pkg/proxy/router/router_test.go

```diff
@@ -173,27 +210,40 @@ func (s *testProxyRouterSuite) InitEnv(c *C) {
 	})
 }
 
-func (s *testProxyRouterSuite) testDialProxy(c *C, addr string) (redis.Conn, error) {
+func (s *testProxyRouterSuite) testMarkOffline(c *C) {
+	suicide := int64(0)
+	proxyMutex.Lock()
+	ss.onSuicide = func() error {
+		atomic.StoreInt64(&suicide, 1)
+		return nil
+	}
+	proxyMutex.Unlock()
+
+	err := models.SetProxyStatus(conn, conf.ProductName, conf.ProxyID, models.PROXY_STATE_MARK_OFFLINE)
+	c.Assert(err, IsNil)
+
+	time.Sleep(3 * time.Second)
+	c.Assert(atomic.LoadInt64(&suicide), Not(Equals), 0)
+}
+
+func (s *testProxyRouterSuite) testDialProxy(c *C, addr string, auth string) redis.Conn {
 	cc, err := redis.Dial("tcp", addr)
 	c.Assert(err, IsNil)
 
-	if len(proxyAuth) > 0 {
-		ok, err := redis.String(cc.Do("AUTH", proxyAuth))
+	if len(auth) > 0 {
+		ok, err := redis.String(cc.Do("AUTH", auth))
 		c.Assert(err, IsNil)
 		c.Assert(ok, Equals, "OK")
 	}
 
-	return cc, nil
+	return cc
 }
 
 func (s *testProxyRouterSuite) TestSingleKeyRedisCmd(c *C) {
-	s.InitEnv(c)
-
-	cc, err := s.testDialProxy(c, "localhost:19000")
-	c.Assert(err, IsNil)
+	cc := s.testDialProxy(c, proxyAddr, proxyAuth)
```

Comment:
testDialProxy func has two usages, first is to get a proxy conn, second is to get a store conn, so we have to add a func variable auth to support it.